src/pytorch_lightning/cli.py

```diff
@@ -478,10 +482,14 @@ def link_optimizers_and_lr_schedulers(parser: LightningArgumentParser) -> None:
                 add_class_path = _add_class_path_generator(class_type)
                 parser.link_arguments(key, link_to, compute_fn=add_class_path)
 
-    def parse_arguments(
-        self, parser: LightningArgumentParser, args: Union[List[str], Dict[str, Any], Namespace]
-    ) -> None:
+    def parse_arguments(self, parser: LightningArgumentParser, args: ArgsType) -> None:
         """Parses command line arguments and stores it in ``self.config``."""
+        if args is not None and len(sys.argv) > 1:
+            raise MisconfigurationException(
```

Comment:
We are moving away from this custom exception class

```suggestion
            raise ValueError(
```